### PR TITLE
fix: decrease error handler verbosity on self recovering errors

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

The subscription sync processor can recover on auth errors by changing to a different auth mode. This change will prevent the `errorHandler` from being called until all auth modes have been attempted.

#### Issue #, if available

N/A

#### Description of how you validated changes

Unit tests.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore/__tests__/subscription.test.ts [hunk 1]
```diff
@@ -566,6 +587,109 @@ describe('sync engine subscription module', () => {
...
+		errorHandler.mockClear();
+		subscriptionProcessor = await instantiateSubscriptionProcessor({
+			errorHandler,
+		});
+	});
+
+	test('error handler once after all retires have failed', done => {
+		const message = PUBSUB_CONTROL_MSG.REALTIME_SUBSCRIPTION_INIT_ERROR;
+		mockObservable = new Observable(observer => {
+			observer.error({
```
Comment (svidgen) [inline packages/datastore/__tests__/subscription.test.ts:605]: Does it make sense to assert that this observer actually generates _more_ errors than we see in `errorHandler`?

**Review by svidgen — APPROVED** [review]

**Review by david-mcafee — APPROVED** [review]
LGTM! 🎉

File: packages/datastore/__tests__/subscription.test.ts (hunk 1 above)
Comment (svidgen) [reply packages/datastore/__tests__/subscription.test.ts:605]: Yes, that might actually be the ideal thing to confirm actually &mdash; that that specific `logger.debug` call is made when `errorHandler` is *not*.

**Comment by svidgen** [comment]
Thanks for additional test coverage -- Looks good!

**Comment by dpilch** [comment]
Not sure where these unit test failures are coming from. Tests are fine on my machine...

**Comment by dpilch** [comment]
Tests are passing now after the latest merge from main.

**Pull request merged**